### DISCUSSION TITLE
Install apache-airflow-providers-apache-hive==6.1.0 error

### DISCUSSION THREAD

**Discussion by tritm94** [apache/airflow#31800] [General] [answered]
### Apache Airflow version

2.6.1

### What happened

Install apache-airflow-providers-apache-hive version 6.1.0 by DockerFile but I got error

### What you think should happen instead

**error: command 'gcc' failed: Permission denied**

### How to reproduce

from airflow.providers.apache.hive.operators.hive import HiveOperator

### Operating System

Python

### Versions of Apache Airflow Providers

2.6.1

### Deployment

Docker-Compose

### Deployment details
This is my DockerFile script:

FROM apache/airflow:2.6.1-python3.8

USER root
RUN apt-get update \
  && apt-get install -y --no-install-recommends \
         openjdk-11-jdk \
  && apt-get autoremove -yqq --purge \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/* docker-context-files/*.whl docker-context-files/*.tar.gz docker-context-files/*.txt || true   

ENV JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64

RUN usermod -g 0 airflow

USER airflow

COPY requirement.txt /requirement.txt

RUN pip3 install --user --upgrade pip
RUN pip install --user --no-cache-dir apache-airflow-upgrade-check

RUN pip install --no-cache-dir --user -r /requirement.txt

RUN pip install --user --no-cache-dir wheel setuptools --upgrade
RUN pip install --user --no-cache-dir "apache-airflow==${AIRFLOW_VERSION}" apache-airflow-providers-apache-hive==6.1.0
RUN pip install --user --no-cache-dir "apache-airflow==${AIRFLOW_VERSION}" apache-airflow-providers-apache-spark==4.1.0
RUN pip install --user --no-cache-dir "apache-airflow==${AIRFLOW_VERSION}" apache-airflow-providers-apache-sqoop

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment, marked as answer]
https://airflow.apache.org/docs/docker-stack/build.html#example-when-you-add-packages-requiring-compilation

**Reply by tritm94** [reply]
I find solution thank you. Same the doc i add 
RUN apt-get update \
  && apt-get install -y --no-install-recommends \
         openjdk-11-jdk build-essential libopenmpi-dev openssh-server openssh-client libsasl2-dev

Install by root user
